include/librealsense2/hpp/rs_options.hpp

```diff
@@ -116,11 +128,23 @@ namespace rs2
             return res > 0;
         }
 
+        /**
+         * sets a callback in case an option in this options container value is updated
+         * \param[in] callback     the callback function
+         */
+        template< class T >
+        void on_options_value_update( T callback ) const
+        {
+            rs2_error * e = nullptr;
+            rs2_set_options_changed_callback_cpp( _options, new option_value_update_callback< T >( callback ), &e );
+            error::handle( e );
+        }
+
         std::vector<rs2_option> get_supported_options()
```

Comment:
Can we change this to return an `options_list`?

Reply:
Affects to many places, will consider doing it in a future PR.
I will open a ticket for it.

Reply:
This is also an API change. Are you sure about this?

Reply:
Should be fairly compatible: we'll return an `options_list` that can be used as a vector and iterated over.
If the unit-tests pass, I think it's a good sign.
You can also easily add a couple of lines to a C++ example and check.
But if you want to keep like this for now, I'm OK with it.